consumers/sender/main.go

```diff
@@ -0,0 +1,273 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"github.com/nsqio/go-nsq"
+	"github.com/sirupsen/logrus"
+
+	"github.com/mailbadger/app/consumers"
+	"github.com/mailbadger/app/emails"
+	"github.com/mailbadger/app/entities"
+	"github.com/mailbadger/app/storage"
+	"github.com/mailbadger/app/storage/redis"
+	"github.com/mailbadger/app/utils"
+)
+
+var (
+	ErrInvalidSesKeys = errors.New("invali ses keys")
+)
+
+const (
+	CacheDuration = 300000 * time.Millisecond
+	CharSet       = "UTF-8"
+)
+
+// MessageHandler implements the nsq handler interface.
+type MessageHandler struct {
+	storage storage.Storage
+	cache   redis.Storage
+}
+
+// HandleMessage is the only requirement needed to fulfill the
+// nsq.Handler interface.
+func (h *MessageHandler) HandleMessage(m *nsq.Message) error {
+	if len(m.Body) == 0 {
+		logrus.Error("Empty message, unable to proceed.")
+		return nil
+	}
+
+	msg := new(entities.SenderTopicParams)
+	err := json.Unmarshal(m.Body, msg)
+	if err != nil {
+		logrus.WithField("body", string(m.Body)).
+			WithError(err).Error("Malformed JSON message.")
+		return nil
+	}
+
+	// check if the message is processing (if the uuid exists in redis that means it is in progress)
+	exist, err := h.cache.Exists(msg.UUID)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"uuid":          msg.UUID,
+			"user_id":       msg.UserID,
+			"campaign_id":   msg.CampaignID,
+			"subscriber_id": msg.SubscriberID,
+		}).WithError(err).Error("Unable to check message existence")
+		return err
+	}
+
+	if exist {
+		return nil
+	}
+
+	if err := h.cache.Set(msg.UUID, m.Body, CacheDuration); err != nil {
```

Comment:
No need to set the whole body here, we only need the key to exist. Also it's a good idea to prefix the keys, for example, instead of `<uuid>` we can write the key as `sender:<uuid>`